src/datachain/lib/dc/datachain.py

```diff
@@ -1761,16 +1766,18 @@ def subtract(  # type: ignore[override]
 
         if on is None and right_on is None:
             other_columns = set(other._effective_signals_schema.db_signals())
-            signals = [
+            common_signals = [
```

Comment:
unrelated changes? AI? cleanup and review and remove?

Reply:
is there a  test for this? can we make it a separate PR with a test? (it is fine to add a test here also)